python/scripts/config_to_json.py

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env python
+"""
+This script converts YAML or Jinja files into JSON for use by kotekan
+
+All files without a ".j2" extension are treated as YMAL, files with a ".j2"
+extension are processed by the Jinja template render.
+
+Extra variables can be passed in with the -e flag, in json/dict format:
+ -e '{"my_val": 20, "my_val2": 30, "my_array": [0,2,1]}'
+   
+The JSON is returned in STDOUT, and error messages are returned in STDERR
+"""
+try:
+    import yaml, json, sys, os, subprocess, errno, argparse
+except ImportError as err:
+    sys.stderr.write(
```

Comment:
There's no guarantee that `sys` is in scope here.

Realistically, the only import that's going to fail for which installing the additional requirements will help is `yaml`.

I'd move the import of everything else (all of which are bundled with Python as part of its standard library) out of the try block.  If the import of any of them fail, then the user's Python is broken in some weird way and it's probably best for them if we just let Python explode (which will happen here anyways if importing `sys` failed.)

Also, I don't know in which order imports are processed.  A missing `yaml` may cause the exception handler to fire before the interpreter has had a chance to consider importing `sys`.

Reply:
Good point, this way I can also improve the error message around this block.